h/accounts/test/views_test.py

```diff
@@ -829,232 +727,280 @@ def test_activate_event_when_validation_succeeds(event,
-    """Check if the user is disabled."""
-    request = Mock(method='POST', authenticated_userid='john')
-    form_validator.return_value = (None, {"username": "john", "pwd": "doe"})
 
-    user = FakeUser(password='abc')
-    user_model.get_by_userid.return_value = user
+@notifications_fixtures
+def test_notifications_404s_if_not_logged_in(authn_policy):
+    request = DummyRequest(post={})
+    authn_policy.authenticated_userid.return_value = None
 
-    profile = ProfileController(request)
-    profile.disable_user()
+    with pytest.raises(httpexceptions.HTTPNotFound):
+        NotificationsController(request).notifications()
 
-    assert user.password == user_model.generate_random_password.return_value
 
+@notifications_fixtures
+def test_notifications_with_invalid_data_returns_form():
+    request = DummyRequest(post={})
+    controller = NotificationsController(request)
+    controller.form = invalid_form()
 
-def test_disable_user_with_no_authenticated_user():
-    exc = ProfileController(Mock(authenticated_userid=None)).disable_user()
+    result = controller.notifications()
 
-    assert isinstance(exc, httpexceptions.HTTPUnauthorized)
+    assert 'form' in result
 
 
-@patch('h.accounts.views.Subscriptions')
-def test_unsubscribe_sets_active_to_False(Subscriptions):
-    """It sets the active field of the subscription to False."""
-    Subscriptions.get_by_id.return_value = Mock(
-        uri='acct:bob@hypothes.is', active=True)
-    request = MagicMock(
-        authenticated_userid='acct:bob@hypothes.is',
-        GET={'subscription_id': 'subscription_id'}
-    )
+@notifications_fixtures
+def test_notifications_form_with_valid_data_updates_subscriptions(authn_policy,
+                                                                  subscriptions_model):
+    request = DummyRequest(post={})
+    authn_policy.authenticated_userid.return_value = 'fiona'
+    subs = [
+        FakeSubscription('reply', True),
+        FakeSubscription('foo', False),
+    ]
+    subscriptions_model.get_subscriptions_for_uri.return_value = subs
+    controller = NotificationsController(request)
+    controller.form = form_validating_to({
+        'notifications': set(['foo'])
+    })
 
-    ProfileController(request).unsubscribe()
+    controller.notifications()
 
-    assert Subscriptions.get_by_id.return_value.active is False
+    assert subs[0].active == False
+    assert subs[1].active == True
 
 
-@patch('h.accounts.views.Subscriptions')
-def test_unsubscribe_not_authorized(Subscriptions):
-    """If you try to unsubscribe someone else's subscription you get a 401."""
-    Subscriptions.get_by_id.return_value = Mock(
-        uri='acct:bob@hypothes.is', active=True)
-    request = MagicMock(
-        authenticated_userid='acct:fred@hypothes.is',
-        GET={'subscription_id': 'subscription_id'}
-    )
+@notifications_fixtures
+def test_notifications_form_with_valid_data_updates_subscriptions(authn_policy,
```

Comment:
There are two tests with the same name `test_notifications_form_with_valid_data_updates_subscriptions`. The second test doesn't appear to actually check if the subscription is updated.